src/analysis/seminorm.lean

```diff
@@ -255,26 +258,51 @@ variables [add_monoid E]
 section has_scalar
 variables [has_scalar 𝕜 E]
 
-instance : inhabited (seminorm 𝕜 E) :=
+instance : has_coe_to_fun (seminorm 𝕜 E) (λ _, E → ℝ) := ⟨λ p, p.to_fun⟩
+
+lemma coe_injective : @function.injective (seminorm 𝕜 E) (E → ℝ) coe_fn
+| ⟨x, _, _⟩ ⟨y, _, _⟩ rfl := rfl
+
+@[ext] lemma ext {p q : seminorm 𝕜 E} (h : (p : E → ℝ) = q) : p = q := coe_injective h
+
+instance : has_zero (seminorm 𝕜 E) :=
 ⟨{ to_fun    := λ _, 0,
-   smul'     := λ _ _, (mul_zero _).symm,
-   triangle' := λ x y, by rw add_zero }⟩
+  smul'     := λ _ _, (mul_zero _).symm,
+  triangle' := λ _ _, by rw add_zero }⟩
 
-instance : has_coe_to_fun (seminorm 𝕜 E) (λ _, E → ℝ) := ⟨λ p, p.to_fun⟩
+@[simp] lemma coe_zero : coe_fn (0 : seminorm 𝕜 E) = 0 := rfl
 
-@[ext] lemma ext {p q : seminorm 𝕜 E} (h : (p : E → ℝ) = q) : p = q :=
-begin
-  cases p,
-  cases q,
-  have : p_to_fun = q_to_fun := h,
-  simp_rw this,
-end
+instance : inhabited (seminorm 𝕜 E) := ⟨0⟩
 
 variables (p : seminorm 𝕜 E) (c : 𝕜) (x y : E) (r : ℝ)
 
 protected lemma smul : p (c • x) = ∥c∥ * p x := p.smul' _ _
 protected lemma triangle : p (x + y) ≤ p x + p y := p.triangle' _ _
 
+lemma mul_sup {a b c : ℝ} (h₁ : 0 ≤ a) : a * (b ⊔ c) = (a * b) ⊔ (a * c) :=
+begin
+  cases le_total b c with h h,
+  { simp [sup_eq_max, max_eq_right h, max_eq_right (mul_le_mul_of_nonneg_left h h₁)] },
+  { simp [sup_eq_max, max_eq_left h, max_eq_left (mul_le_mul_of_nonneg_left h h₁)] },
+end
```

Comment:
@YaelDillies, you posted a generalized version of this on Zulip, right? What file should it live in?

Reply:
`algebra.order.ring`. It's its only import.

Reply:
there is a `max_mul_of_nonneg` in `algebra.order.ring`

Reply:
Great! Then use it 😄